src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp

```diff
@@ -314,6 +314,19 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
             _kernel_data.kernels[i].code.kernelString.reset();
         }
     }
+
+    void set_kernels(std::map<const std::string, kernel::ptr>& kernels) override {
```

Comment:
Can't we do this with combination of existing methods? 
(set_kernel_ids + init_kernels)
or migrate with them?